fleetconfig-controller/api/v1alpha1/validation.go

```diff
@@ -174,6 +177,62 @@ func validateAddons(newObject *FleetConfig) field.ErrorList {
 	return errs
 }
 
+// validateHubAddons validates HubAddOn configurations and usage
+func validateHubAddons(ctx context.Context, oldObject, newObject *FleetConfig) field.ErrorList {
+	errs := field.ErrorList{}
+
+	// Check for name clashes between HubAddOns and AddOnConfigs
+	addOnConfigNames := make(map[string]struct{})
+	for _, ac := range newObject.Spec.AddOnConfigs {
+		addOnConfigNames[ac.Name] = struct{}{}
+	}
+
+	for i, ha := range newObject.Spec.HubAddOns {
+		if _, found := addOnConfigNames[ha.Name]; found {
+			errs = append(errs, field.Invalid(field.NewPath("hubAddOn").Index(i), ha.Name,
+				fmt.Sprintf("hubAddOn name %s clashes with an existing addOnConfig name", ha.Name)))
+		}
+	}
+
+	// Check if any removed hub addons are still in use by managed cluster addons
+	if oldObject != nil {
+		oldHubAddOns := make(map[string]struct{})
+		for _, ha := range oldObject.Spec.HubAddOns {
+			oldHubAddOns[ha.Name] = struct{}{}
+		}
+
+		newHubAddOns := make(map[string]struct{})
+		for _, ha := range newObject.Spec.HubAddOns {
+			newHubAddOns[ha.Name] = struct{}{}
+		}
+
+		removedHubAddOns := make([]string, 0)
+		for name := range oldHubAddOns {
+			if _, found := newHubAddOns[name]; !found {
+				removedHubAddOns = append(removedHubAddOns, name)
+			}
+		}
+
+		// Check if any removed hub addons are still in use by managed cluster addons
+		if len(removedHubAddOns) > 0 {
+			mcAddOns, err := getManagedClusterAddOns(ctx)
+			if err != nil {
+				errs = append(errs, field.InternalError(field.NewPath("hubAddOn"), err))
+			} else {
+				// Check if any removed hub addons are still in use
+				for _, removedHubAddOn := range removedHubAddOns {
+					if isAddondEnabled(mcAddOns, removedHubAddOn) {
+						errs = append(errs, field.Invalid(field.NewPath("hubAddOn"), removedHubAddOn,
+							fmt.Sprintf("cannot remove hubAddOn %s as it is still in use by managedclusteraddons", removedHubAddOn)))
```

Comment:
please collect all relevant removedHubAddOns and add the err at the end with a slice of addon names. otherwise this err should not be plural. one err with a slice is more concise though.